Solution1/Solution1.Tests/BL/Services/WeatherServiceTests.cs

```diff
@@ -168,37 +175,79 @@ public async Task GetWeatherByArrayCityNameAsync_ReturnedDictionaryForWeatherRes
             var result = await _weatherService.GetWeatherByArrayCityNameAsync(listCityName);
 
             // Assert
-            Assert.Contains(result.Keys, k => k == true);
-            Assert.Contains(result.Keys, k => k == false);
-
-            Assert.Equal(2, result[true].Count());
-            Assert.Equal(3, result[false].Count());
-
-            Assert.Contains(result[true], weatherResponse => weatherResponse.CityName == cityName);
-            Assert.Contains(result[true], weatherResponse => weatherResponse.CityName == cityName2);
-            Assert.Contains(result[false], weatherResponse => weatherResponse.CityName == cityName3 
-                                                              && weatherResponse.ErrorMessage == errorMessage3); 
-            Assert.Contains(result[false], weatherResponse => weatherResponse.CityName == cityName4 
+            Assert.Contains(result.Keys, k => k == ResponseStatus.Successful);
+            Assert.Contains(result.Keys, k => k == ResponseStatus.Fail);
+            Assert.DoesNotContain(result.Keys, k => k == ResponseStatus.Canceled);
```

Comment:
is there any reason why for Canceled case separate test? your method can return either successful/failed, either canceled? they cannot exist together?

Reply:
+